services/garage-backend/src/routers/thingRouter.ts

```diff
@@ -1,21 +1,55 @@
 import express from 'express';
+import multer from 'multer';
+import * as fs from 'fs';
 
-import { requireAuth } from '../middlewares/auth';
 import {
   createThing,
   findThingById,
   updateThing,
   findAllThings,
   deleteThing,
+  restoreThing,
 } from '../controllers/thingController';
+import { requireAuth } from '../middlewares/auth';
 
 const router = express.Router();
 router.use(requireAuth());
 
+const storage = multer.diskStorage({
+  destination(req, file, cb) {
+    const dir = './uploads/';
+    fs.exists(dir, (exist) => {
+      if (!exist) {
+        return fs.mkdir(dir, (error) => cb(error, dir));
+      }
+      return cb(null, dir);
+    });
+  },
+
+  filename(req: any, file: any, cb: any) {
+    const filename = file.originalname;
+    const fileExtension = filename.split('.')[1];
```

Comment:
if there's more than one dot in a filename this will not work. I suggest to change it to this
```suggestion
    const filenameParts = filename.split('.')
    const fileExtension = filenameParts[filenameParts.length - 1];
```